### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ciricode**

### CODE REVIEW

**Comment by ging-dev** [comment]
i tested it, the result can be around 120k req/s, not too bad.

**Comment by ging-dev** [comment]
Should add usage in `README.md`

**Comment by ciricode** [comment]
> Should add usage in `README.md`

i will add it

File: utils.go [hunk 1]
```diff
@@ -3,10 +3,9 @@ package main
 import (
 	"bufio"
 	"crypto/tls"
-	"fmt"
 	"math/rand"
-	"net"
-	"net/url"
+	"net/http"
+	xurl "net/url"
```
Comment (ging-dev) [inline utils.go]: used `u, err :=...`

**Comment by ging-dev** [comment]
please rebase...

**Comment by ciricode** [comment]
> please rebase...

done!

**Review by ging-dev — APPROVED** [review]

**Comment by ging-dev** [comment]
Thanks!

**Pull request merged**